src/org/zaproxy/zap/extension/api/CoreAPI.java

```diff
@@ -453,6 +460,22 @@ public void process(HttpMessage msg) {
 			Stats.clear(this.getParam(params, PARAM_KEY_PREFIX, ""));
 			return ApiResponseElement.OK;
 			
+		} else if (ACTION_DELETE_SITE_NODE.equals(name)) {
+			try {
+				String url = params.getString(PARAM_URL);
+				String method = getParam(params, PARAM_METHOD, "GET");
```

Comment:
OK, let it be GET but then we need to change an if statement in [`SiteMap.getLeafName(String, URI, String, String)`](https://github.com/zaproxy/zaproxy/blob/a48e956541defb4d5124fb83821a332900d4a1c0/src/org/parosproxy/paros/model/SiteMap.java#L549) to also consider an empty `String` as no method.
So, when the user specifies an empty method (as opposed to no method, which defaults to GET) it's able to find the leaf (directory) nodes.